bears/bearController.js

```diff
@@ -1,24 +1,83 @@
 const router = require('express').Router();
 
+const Bear = require('./bearModel');
+
 router
   .route('/')
   .get((req, res) => {
-    res.status(200).json({ route: '/api/bears/' });
+    Bear
+    .find()
+    .then(bears => {
+      res.status(200).json(bears);
+    })
+    .catch(err => res.status(500).json({error: "There was an error fetching the data."}));
   })
   .post((req, res) => {
-    res.status(201).json({ status: 'please implement POST functionality' });
+    const {species, latinName} = req.body;
+    const newBear = new Bear ({species, latinName});
+    if (!species || !latinName){
+      res.status(400).json({error: "Please provide both species and latinName."})
+    }
+    newBear
+    .save()
+    .then(savedBear => {
+      res.status(201).json(savedBear);
+    })
+    .catch(err => {
+      res.status(500).json({error: "There was an error while saving to the db."})
+    });
+    // res.status(201).json({ status: 'please implement POST functionality' });
   });
 
 router
   .route('/:id')
   .get((req, res) => {
-    res.status(200).json({ route: '/api/bears/' + req.params.id });
+    // res.status(200).json({ route: '/api/bears/' + req.params.id });
+    const {id} = req.params;
+    if(id.length !== 24){
+      res.status(400).json({error: "IDs need to be 24 characters long."})
+    }
+    Bear.findById(id)
+    .then(bear => {
+      if(bear === null){
+        res.status(404).json({error: "Bear cannot be found with given ID."});
+        return;
+      } else {
+        console.log(bear.id.length);
+        res.status(200).json(bear);
+      }
+    })
+    .catch(err => {
+      res.status(500).json({error: 'Something has gone wrong, please try again.'});
+    })
   })
   .delete((req, res) => {
-    res.status(200).json({ status: 'please implement DELETE functionality' });
+    // res.status(200).json({ status: 'please implement DELETE functionality' });
+    const {id} = req.params;
+    console.log(id);
+    if (id.length !== 24){
+      res.status(400).send({error: "Your ID needs to be 24 characters long."})
+      return;
+    }
+    Bear.findByIdAndRemove(id)
```

Comment:
I believe we also need the upsert parameter here, please review the api information that I DM'd you.